### PR TITLE
Unbreak master

### PR DESCRIPTION
**Pull request by thofma**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ Nemo = "0.23.0"
 RandomExtensions = "0.4.2"
 Singular_jll = "~402.000.102"
 julia = "1.3"
-libsingular_julia_jll = "~0.9.200"
+libsingular_julia_jll = "~0.10.0"
```
Comment (thofma) [inline Project.toml:26]:
```suggestion
libsingular_julia_jll = "~0.11.0"
```

**Comment by tthsqe12** [comment]
I can reproduce the error on 1.5 locally by both
1. using the libsingular_julia_jll v0.11.5+0, and
2. using the override but having julia 1.6 build the binaries

If julia 1.5 builds the binaries in the override, then 1.5 is happy.

**Comment by thofma** [comment]
We are having problems with the libsingular_julia binaries we have built here: https://github.com/JuliaPackaging/Yggdrasil/pull/2974
Apparently there are some undefined symbols: https://github.com/oscar-system/Singular.jl/pull/430/checks?check_run_id=2574651899.

@fingolfin @benlorenz Do you have a clue what the error could be? Strangley enough, the CI tests of https://github.com/oscar-system/libsingular-julia itself were passing when we merged the PRs. But now with the binaries provided by Yggdrasil it is failing.

**Comment by benlorenz** [comment]
This seems to be caused by different libcxxwrap_julia versions:
The Yggdrasil build used `libcxxwrap_julia_jll v0.8.6+0` which contains libcxxwrap_julia 0.8.3.
But in the tests (e.g. for julia 1.5) we get `libcxxwrap_julia_jll v0.8.5+1` which contains libcxxwrap_julia 0.8.2.
Both these libcxxwrap versions do have the same soname for the library but they are incompatible, e.g. regarding that undefined symbol (`_ZN5jlcxx10apply_typeEP11_jl_value_tP14_jl_datatype_t == jlcxx::apply_type(_jl_value_t*, _jl_datatype_t*)`) that was introduced in 0.8.3.

The dependencies during the build are not really resolved properly with regard to the julia compat as 0.8.6 should not be used for julia <1.6 but it used for the build for julia 1.3. That may be a (known?) bug in BinaryBuilder.

We might try to pin the libcxxwrap_julia_jll version to 0.8.3 for all builds but I do not know what else might break then (similar to the fixed gmp / mpfr versions).
Or we can try to create releases for libcxxwrap_julia_jll for julia 1.3-1.5 that are based on the 0.8.3 release instead of 0.8.2.

Maybe I can come up with some other options when I have some more time later or tomorrow.

**Comment by benlorenz** [comment]
Looking into this a bit more, I found the `julia_version` argument to the `Platform` type which is probably what we need here, see e.g. https://github.com/JuliaPackaging/BinaryBuilderBase.jl/commit/cab650c82efd15cf5676bc05b553c5fbcb7f1054#diff-7191d1cf5dee1b5b6cdf1dbe4abba72c5880caa2f3f4a8f3d5c0726b07a3259bR108

So we could try adding the `julia_version` as a keyword argument to all `Platform` objects which should resolve with the correct julia version then. Then we might not need the version restriction for gmp / mpfr anymore.

**Comment by thofma** [comment]
Where would we do this? In the `build_tarballs.jl`? At the moment we are doing there: https://github.com/JuliaPackaging/Yggdrasil/blob/9fbc2ca069c19c6a6fc2bad285c5f5d0774117cb/L/libsingular_julia/common.jl#L35. But this is not setting the `julia_version`?

**Comment by benlorenz** [comment]
No, `julia_platforms` doesn't contain the `julia_version` as a tag. Maybe it should?
I really don't know if this will break other stuff (e.g. I don't know which julia version support these tags in the triplets...), but we could at least get rid of all those explicit gmp / mpfr versions then. Maybe @fingolfin has some more insight?

We can try adding it manually like here: https://github.com/benlorenz/Yggdrasil/commit/39c015b8febc8746c201ba37629abb6e411eb3c3
I am currently running a local build with that and want to test the binaries.

**Comment by tthsqe12** [comment]
How is your local build and testing doing?

The deeper issue here is that: while the CI for libsingular_julia_jll told us the lights were all green, they were in fact red for < 1.6. Is it possible for the libsingular_julia_jll CI to reproduce the experience that we are going to end up with after passing everything through yggdrasil?

**Comment by fingolfin** [comment]
So @benlorenz attempts to fix this by setting `julia_version` in the platform. This would be the right fix (and I would have used it before), *except* that (IIRC) the resulting JLLs will only be loadable in Julia >= 1.6 (as the Julia version ends up in the platform name string -- which is what we want, but Julia <= 1.5 chokes on it and dies).

So I think the best we can do is to set `julia_version` iff we are building for Julia >= 1.5. But will that help us? Not sure.

In case of `libcxxwrap_julia`, deriving the "right" version for each target Julia is not so hard, so we could hardcode the version dependency on it; but that also has drawbacks (such as: we must remember to increment it whenever `libcxxwrap_julia` is updated. This could also be seen as a bonus, though: we want to be in full control over all cxxwrap updates, as in the past they often introduced breakages for us.

**Comment by benlorenz** [comment]
> So @benlorenz attempts to fix this by setting `julia_version` in the platform. This would be the right fix (and I would have used it before), _except_ that (IIRC) the resulting JLLs will only be loadable in Julia >= 1.6 (as the Julia version ends up in the platform name string -- which is what we want, but Julia <= 1.5 chokes on it and dies).

That is (unfortunately) correct, my experiments produce such an error for 1.3 and 1.5:
```
┌ Warning: Platform `aarch64-linux-gnu-cxx11-julia_version+1.5` is not an officially supported platform
```

> In case of `libcxxwrap_julia`, deriving the "right" version for each target Julia is not so hard, so we could hardcode the version dependency on it; but that also has drawbacks (such as: we must remember to increment it whenever `libcxxwrap_julia` is updated. This could also be seen as a bonus, though: we want to be in full control over all cxxwrap updates, as in the past they often introduced breakages for us.

Yes, lets do this, at least until we can drop support for <1.6.


> The deeper issue here is that: while the CI for libsingular_julia_jll told us the lights were all green, they were in fact red for < 1.6. Is it possible for the libsingular_julia_jll CI to reproduce the experience that we are going to end up with in singular.jl after passing everything through yggdrasil?

Well ... there was nothing wrong with the pull request and there is nothing we can do here to fix this. Also this would have happened as well if we just needed to rebuild libsingular_julia_jll on yggdrasil for any other reason.

It will always be possible that changes in libcxxwrap or yggdrasil will break this, in this case the main problem is that there are libcxxwrap_julia_jll 0.8.3, 0.8.4, 0.8.5 for julia versions 1.3, 1.4 and 1.5 which are built from the same libcxxwrap release. But libcxxwrap_julia_jll 0.8.6 for julia 1.6 was created from a different libcxxwrap which is incompatible with the other release.
(This was not really how I thought these 0.8.x releases were supposed to be organized)

We might try to build libsingular_julia with binarybuilder + yggdrasil and deploy them locally in one CI step, then transfer the results to the other steps and try to reuse them there. But this will definitely get messy ...

**Comment by fingolfin** [comment]
Indeed, that libcxxwrap 0.8.6 has a different source version is IMHO a bug there and should be fixed -- e.g. by making libcxxwrap 0.8.1x releases, with x=3,4,5,6 (or 0.9.x if there are other changes to warrant it...)

**Comment by thofma** [comment]
I will try to adjust the script on Yggdrasil to work around the bug.

**Comment by thofma** [comment]
Thanks y'all.

**Comment by tthsqe12** [comment]
Yes. Thanks yall.

**Pull request merged**